test/analysis/buildTypeWeights.test.ts

```diff
@@ -0,0 +1,239 @@
+            schema = buildSchema(`
+                type Test {
+                    num: Int,
+                    id: ID,
+                    float: Float,
+                    bool: Boolean,
+                    string: String
+                } 
+            `);
+
+            expect(buildTypeWeightsFromSchema(schema)).toEqual({
+                Test: {
+                    weight: 1,
+                    fields: {
+                        num: 0,
+                        id: 0,
+                        float: 0,
+                        bool: 0,
+                        string: 0,
+                    },
+                },
+            });
+        });
+    });
+
+    describe('changes "type weight object" type weights with user configuration of...', () => {
+        let expectedOutput: TypeWeightObject;
+
+        beforeEach(() => {
+            schema = buildSchema(`
+                type Query {
+                    user: User
+                    movie: Movie
+                }
+                
+                type User {
+                    name: String
+                    email: String
+                }
+                
+                type Movie {
+                    name: String
+                    director: User
+                }   
+            `);
+
+            // This expected output is using default type weight settings.
+            // Each test will override values for feild weights configuration.
+            expectedOutput = {
+                Query: {
+                    weight: 1,
+                    fields: {},
+                },
+                User: {
+                    weight: 1,
+                    fields: {
+                        name: 0,
+                        email: 0,
+                    },
+                },
+                Movie: {
+                    weight: 1,
+                    fields: {
+                        name: 0,
+                    },
+                },
+            };
+        });
+
+        // this is only if we choose to have 'query' as its own property (seperate from object types) in the user configuration options
+        xtest('query parameter', () => {
+            const typeWeightObject = buildTypeWeightsFromSchema(schema, {
+                query: 2,
+            });
+            expectedOutput.query.weight = 2;
+
+            expect(typeWeightObject).toEqual({ expectedOutput });
+        });
+
+        test('object parameter', () => {
+            const typeWeightObject = buildTypeWeightsFromSchema(schema, {
+                object: 2,
+            });
+
+            expectedOutput.user.weight = 2;
+            expectedOutput.movie.weight = 2;
+
+            expect(typeWeightObject).toEqual({ expectedOutput });
+        });
+
+        test('scalar parameter', () => {
+            const typeWeightObject = buildTypeWeightsFromSchema(schema, {
+                scalar: 2,
+            });
+
+            expectedOutput.user.fields.name = 2;
```

Comment:
Similar to the previous comment, should we also check that Object types still have the expected weight?

Reply:
Similar to previous comment, the objects will still have the expected weight of one as declared on line 148.